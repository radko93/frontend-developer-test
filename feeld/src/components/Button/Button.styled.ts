import styled from 'styled-components/native'
import { responsivePercentageFontSize, responsiveDPSize } from '../../utils/responsive'

export const Container = styled.TouchableOpacity<{ color: string }>`
  border-color: ${({ color }) => color};
  border-width: 1;
  border-radius: 50;
  justify-content: center;
  align-items: center;
  padding-vertical: ${responsiveDPSize(5)};
  padding-horizontal: ${responsiveDPSize(10)};
`

export const Label = styled.Text<{ color: string }>`
  color: ${({ color }) => color};
  font-size: ${responsivePercentageFontSize(2)};
`
