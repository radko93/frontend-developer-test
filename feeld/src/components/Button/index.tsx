import React from 'react'
import { Container, Label } from './Button.styled'

type Props = {
  onPress: () => void
  label: string
  color: string
}

const Button = ({ onPress, color, label }: Props): React.ReactElement => {
  return (
    <Container color={color} onPress={onPress} testID="button">
      <Label color={color}>{label}</Label>
    </Container>
  )
}

export default Button
