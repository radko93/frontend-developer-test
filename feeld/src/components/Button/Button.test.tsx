import React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import Button from '.'

const props = {
  onPress: jest.fn(),
  label: 'Test',
  color: '#00F',
}

const wrapperComponent = <Button {...props} />

describe('Button Component', (): void => {
  it('Renders Button with default props', (): void => {
    const wrapper = render(wrapperComponent)
    expect(wrapper).toMatchSnapshot()
  })
  it('Should fire onPress event', (): void => {
    const wrapper = render(wrapperComponent)
    const { getByTestId } = wrapper
    fireEvent.press(getByTestId('button'))
  })
})
