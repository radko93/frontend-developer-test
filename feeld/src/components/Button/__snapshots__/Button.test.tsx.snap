// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renders Button with default props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#00F",
      "borderRadius": 50,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 15,
      "paddingVertical": 8,
    }
  }
  testID="button"
>
  Test
</View>
`;
